pkg/kubelet/volume/empty_dir/empty_dir.go

```diff
@@ -0,0 +1,126 @@
+/*
+Copyright 2014 Google Inc. All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package empty_dir
+
+import (
+	"fmt"
+	"os"
+	"path"
+
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/api"
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/kubelet/volume"
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/types"
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/util"
+)
+
+func init() {
+	volume.RegisterPlugin(&emptyDirPlugin{nil, false})
```

Comment:
I would suggest inverting this model to be declarative:

```
func Plugin() PluginInterface {
}
```

Then in cmd/kubelet/plugins.go do

```
func VolumePlugins() []Plugins {
  return []Plugins{
    emptydir.Plugin(),
  }
}
```

and use it directly.  We went with this magic static variable initialization originally for geard, and the more I worked with it the more frustrated I was about testing it.  You don't really need volume.RegisterPlugin in that model, you just pass the Kubelet a list of plugins.